docs: fix incorrect file path (#4368)
backstage/community-plugins: docs/plugin-maintainers-guide.md

## Opt-in to Knip Reports Check

> Plugin owners can opt in to Knip reports check in CI by creating a `bcp.json` file in the root of their workspace (`workspaces/${WORKSPACE}/bcp.json`) and adding `"knip-reports": true`. This ensures that knip reports in your workspace stay up to date.

Reviewer: perhaps this is a bit more clear

```suggestion
Plugin owners can opt in to Knip reports check in CI by creating a `bcp.json` file in the root of their workspace (`workspaces/${WORKSPACE}/bcp.json`) with the content `{ "knip-reports": true }`. This ensures that knip reports in your workspace stay up to date.
```
